File: HealthSysHub.Web.UI/Services/PharmacyOrderRequestService.cs
using HealthSysHub.Web.UI.Factory;
using HealthSysHub.Web.UI.Interfaces;
using HealthSysHub.Web.Utility.Models;
using System;

namespace HealthSysHub.Web.UI.Services
{
    public class PharmacyOrderRequestService : IPharmacyOrderRequestService
    {
        private readonly IRepositoryFactory _repository;

        public PharmacyOrderRequestService(IRepositoryFactory repository)
        {
            _repository = repository;
        }

        public async Task<PharmacyOrderRequestDetails> GetPharmacyOrderRequestDetailAsync(Guid pharmacyOrderRequestId)
        {
            string url = Path.Combine("GetPharmacyOrderRequestDetailAsync", pharmacyOrderRequestId.ToString());

            return await _repository.SendAsync<PharmacyOrderRequestDetails>(HttpMethod.Get, url);
        }

        public async Task<List<PharmacyOrderRequestDetails>> GetPharmacyOrderRequestsAsync()
        {
            return await _repository.SendAsync<List<PharmacyOrderRequestDetails>>(HttpMethod.Get, "GetPharmacyOrderRequestsAsync");
        }

        public async Task<List<PharmacyOrderRequestDetails>> GetPharmacyOrderRequestsByHospitalAsync(Guid hospitalId)
        {
            string url = Path.Combine("GetPharmacyOrderRequestsByHospitalAsync", hospitalId.ToString());

            return await _repository.SendAsync<List<PharmacyOrderRequestDetails>>(HttpMethod.Get, url);
        }

        public async Task<List<PharmacyOrderRequestDetails>> GetPharmacyOrderRequestsByPatientAsync(Guid patientId)
        {
            string url = Path.Combine("GetPharmacyOrderRequestsByPatientAsync", patientId.ToString());

            return await _repository.SendAsync<List<PharmacyOrderRequestDetails>>(HttpMethod.Get, url);
        }

        public async Task<List<PharmacyOrderRequestDetails>> GetPharmacyOrderRequestsByPharmacyAsync(Guid pharmacyId)
        {
            string url = Path.Combine("GetPharmacyOrderRequestsByPharmacyAsync", pharmacyId.ToString());

            return await _repository.SendAsync<List<PharmacyOrderRequestDetails>>(HttpMethod.Get, url);
        }

        public async Task<PharmacyOrderRequestDetails> InsertOrUpdatePharmacyOrderRequestDetailsAsync(PharmacyOrderRequestDetails requestDetails)
        {
            return await _repository.SendAsync<PharmacyOrderRequestDetails, PharmacyOrderRequestDetails>(HttpMethod.Post, "PharmacyOrderRequest/InsertOrUpdatePharmacyOrderRequestAsync", requestDetails);

        }
    }
}

File: HealthSysHub.Web.Pharmacy.UI/src/app/services/pharmacy-order-request-service.ts
import { Injectable } from "@angular/core";
import { ApiService } from "./apiservice.service";
import { Observable } from "rxjs";
import { environment } from "../../environment";
import { PharmacyOrderRequestDetails } from "../models/pharmacyorderrequestdetails";
import { ProcessPharmacyOrderRequest } from "../models/processpharmacyorderrequest";
import { ProcessPharmacyOrderRequestResponse } from "../models/processpharmacyorderrequestresponse";

@Injectable({
    providedIn: "root"
})
export class PharmacyOrderRequestService {
    constructor(private apiService: ApiService) { }

    GetPharmacyOrderRequestsByPharmacyAsync(pharmacyId: string): Observable<PharmacyOrderRequestDetails[]> {
        const url = `${environment.UrlConstants.GetPharmacyOrderRequestsByPharmacyAsync}/${pharmacyId}`;
        return this.apiService.send<PharmacyOrderRequestDetails[]>("GET", url);
    }
    GetPharmacyOrderRequestsByHospitalAsync(hospitalId: string): Observable<PharmacyOrderRequestDetails[]> {
        const url = `${environment.UrlConstants.GetPharmacyOrderRequestsByHospitalAsync}/${hospitalId}`;
        return this.apiService.send<PharmacyOrderRequestDetails[]>("GET", url);
    }
    GetPharmacyOrderRequestsByPatientAsync(patientId: string): Observable<PharmacyOrderRequestDetails[]> {
        const url = `${environment.UrlConstants.GetPharmacyOrderRequestsByPatientAsync}/${patientId}`;
        return this.apiService.send<PharmacyOrderRequestDetails[]>("GET", url);
    }
    GetPharmacyOrderRequestDetailAsync(pharmacyOrderRequestId: string): Observable<PharmacyOrderRequestDetails> {
        const url = `${environment.UrlConstants.GetPharmacyOrderRequestDetailAsync}/${pharmacyOrderRequestId}`;
        return this.apiService.send<PharmacyOrderRequestDetails>("GET", url);
    }
    InsertOrUpdatePharmacyOrderRequestAsync(pharmacyOrderReques: PharmacyOrderRequestDetails): Observable<PharmacyOrderRequestDetails> {
        return this.apiService.send<PharmacyOrderRequestDetails>('POST', environment.UrlConstants.InsertOrUpdatePharmacyOrderRequestAsync, pharmacyOrderReques);
    }
    ProcessPharmacyOrderRequestAsync(requestDetails: ProcessPharmacyOrderRequest): Observable<ProcessPharmacyOrderRequestResponse> {
        return this.apiService.send<ProcessPharmacyOrderRequestResponse>('POST', environment.UrlConstants.ProcessPharmacyOrderRequestAsync, requestDetails);
    }
}